### PR TITLE
[WFLY-12763] Create a 'bean-validation' layer separate from 'cdi'

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.jboss.org/browse/WFLY-12763

I also did a bit of doc editing.

I expect to need to rebase this once #12775 and #12776

### CODE REVIEW

**Comment by rhusar** [comment]
Needs a rebase now (on documentation... sigh).

**Comment by bstansberry** [comment]
@rhusar Done. I expected the rebase. :)

File: galleon-pack/src/main/resources/layers/standalone/bean-validation/layer-spec.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="cdi">
```
Comment (kabir) [inline galleon-pack/src/main/resources/layers/standalone/bean-validation/layer-spec.xml]: This name should probably be 'bean-validation'?

Comment (bstansberry) [reply galleon-pack/src/main/resources/layers/standalone/bean-validation/layer-spec.xml]: @kabir Fixed; thanks!

**Pull request merged**